controllers for GET

Amithabh-A opened the pull request:
file contains controllers for Announcements. After needed debugs, same code can be used for other models.
Also, models are not created. If requested, I can write them.

Amithabh-A committed:
Added controller.js

file contains controllers for Announcements. After needed debugs, same code can be used for other models.

SyedMa3 commented on src/backend/controllers/GETcontroller.js:
Separate files for each model. And rename the file as e.g. `announcements.controllers.js`
Use `express-async-handler`.
Contact routes team to decide the name for controller functions.
Write the remaining controllers too.

Amithabh-A committed:
just before rewrite controllers

I am not writing models. I am going to write controllers. Let the models team figure it out how to handle model stuffs

Amithabh-A committed:
controllers for threads and clubs.

Amithabh-A committed:
comments added.

SyedMa3 commented on src/backend/controllers/clubs.controllers.js:
Give proper names to functions. Refer to Neeraj's controller names.

Amithabh-A replied:
done.

Amithabh-A committed:
Fixed function names and file namings.

Amithabh-A committed:
more fixes in clubs.

SyedMa3 committed:
Merge branch 'main' into main
